app/views/rubygems/_dependencies.html.erb

```diff
@@ -3,18 +3,14 @@
     <h3 class="t-list__heading"><%= t '.header', :title => dependencies.first.scope.titleize %> (<%= dependencies.size %>):</h3>
     <div class="t-list__items">
       <% dependencies.each do |dependency| %>
-        <% if rubygem = dependency.rubygem %>
-          <li class="gem__requirement-wrap">
-            <%= link_to rubygem_path(rubygem), :class => 't-list__item' do %>
-              <strong><%= rubygem.name %></strong>
-            <% end %>
-            <%= dependency.clean_requirements %>
-          </li>
-        <% elsif dependency&.name %>
-          <p class="t-list__item gem__unregistered" title="unregistered gem">
-            <strong><%= dependency.name %></strong> <%= dependency.clean_requirements %>
-          </p>
-        <% end %>
+        <%- rubygem = dependency.rubygem -%>
```

Comment:
To avoid variable assignment in templates you can also use `tap`.

```erb
<% dependency.rubygem.tap do |rubygem| %>
  ...
<% end %>
```

Reply:
Agree!